src/Compilers/CSharp/Test/Syntax/Syntax/SyntaxTests.cs

```diff
@@ -276,5 +279,54 @@ public void IsAttributeTargetSpecifier()
                 }
             }
         }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/72300")]
+        public void TestAllKindsReturnedFromGetKindsMethodsExist()
+        {
+            foreach (var method in typeof(SyntaxFacts).GetMethods(BindingFlags.Public | BindingFlags.Static))
+            {
+                if (method.ReturnType == typeof(IEnumerable<SyntaxKind>) && method.GetParameters() is [])
+                {
+                    foreach (var kind in (IEnumerable<SyntaxKind>)method.Invoke(null, null))
+                    {
+                        Assert.True(Enum.IsDefined(typeof(SyntaxKind), kind), $"Nonexistent kind '{kind}' returned from method '{method.Name}'");
+                    }
+                }
+            }
+        }
+
+        [Theory, WorkItem("https://github.com/dotnet/roslyn/issues/72300")]
+        [InlineData(nameof(SyntaxFacts.GetContextualKeywordKinds), SyntaxKind.YieldKeyword, SyntaxKind.FileKeyword)]
+        [InlineData(nameof(SyntaxFacts.GetPreprocessorKeywordKinds), SyntaxKind.ElifKeyword, SyntaxKind.RestoreKeyword)] // This case doesn't cover all possible preprocessor kinds
```

Comment:
>// This case doesn't cover all possible preprocessor kinds [](http://example.com/codeflow?start=121&length=58)

Then, perhaps, it should have its own dedicated test that provides proper coverage?